ca_code/scripts/run_gen_texmean.py

```diff
@@ -0,0 +1,121 @@
+    assets = AttrDict(train_dataset.static_assets)
+    geo_fn = GeometryModule(
+        th.LongTensor(assets.topology.vi),
+        assets.topology.vt,
+        assets.topology.vti,
+        assets.topology.v2uv,
+        uv_size=1024,
+        impaint=True,
+    ).to(device)
+    lbs_fn = LBSModule(
+        assets.lbs_model_json,
+        assets.lbs_config_dict,
+        assets.template_mesh_unscaled[None],
+        assets.skeleton_scales,
+        global_scaling=[10.0, 10.0, 10.0],  # meter
+    ).to(device)
+
+    rl = RenderLayer(
+        h=config.model.renderer.image_height,
+        w=config.model.renderer.image_width,
+        vt=geo_fn.vt,
+        vi=geo_fn.vi.int(),
+        vti=geo_fn.vti,
+        flip_uvs=False,
+    ).to(device)
+
+    uv_size = (1024, 1024)
+    inpaint_threshold = 100.0
+    vt = geo_fn.vt
+    vi = geo_fn.vi
+    vti = geo_fn.vti
+    index_image = make_uv_vert_index(
+        vt, vi, vti, uv_shape=uv_size, flip_uv=True
+    ).cpu()
+    face_index, bary_image = make_uv_barys(vt, vti, uv_shape=uv_size, flip_uv=True)
+
+    # inpaint index uv images
+    index_image, bary_image = index_image_impaint(
+                    index_image, bary_image, inpaint_threshold
+                )
+    face_index = index_image_impaint(face_index, distance_threshold=inpaint_threshold)
+
+    frame_list = train_dataset.get_frame_list(
+        fully_lit_only=train_dataset.fully_lit_only,
+        partially_lit_only=train_dataset.partially_lit_only,
+    )
+    # we suggest to use occlusion-free 5 consecutive frames for tex_mean extraction
+    frame_list = frame_list[:5]
+    camera_list = train_dataset.get_camera_list()
+    tex_total = th.zeros(1, 3, 1024, 1024).float().to(device)
+    tex_cnt = th.zeros(1, 3, 1024, 1024).float().to(device)
+    for fid in frame_list:
+        for cid in tqdm(camera_list):
+            try:
+                current_data = train_dataset.get(fid, cid)
+            except:
+                logger.warning(f"error when loading cam{cid} frame{fid}, skipping")
+                continue
+            current_data = to_device(current_data, device)
+            img = current_data['image'][None].float()
+            extrin = current_data['Rt'][None]
+            intrin = current_data['K'][None]
+            pose = current_data['pose'][None]
+            mesh_world = lbs_fn.pose(th.zeros_like(lbs_fn.lbs_template_verts), pose)
+            faces = geo_fn.vi
+            ply = (mesh_world, faces)
+            tex_img, tex_mask = get_tex_rl(rl, img, ply, extrin, intrin, face_index, index_image, bary_image)
+
+            tex_total += tex_img
+            tex_cnt += tex_mask.float()
+
+    tex_total /= tex_cnt + 1e-5
+    tex_mean = th.flip(tex_total[0].permute(1, 2, 0), (0,)).cpu().numpy()
+    tex_mean_path = os.path.join(config.data.root_path, "uv_image", "color_mean.png")
+    cv2.imwrite(tex_mean_path, tex_mean[..., [2, 1, 0]])
+
+if __name__ == "__main__":
```

Comment:
Consider adding a check on whether the images already exists, and if they do then don't run the computation; then you can call this anytime people train and people don't need to worry about a separate pipeline.

Reply:
The instruction is updated in the README clearly. And I feel the best practice would be release `color_mean.png` for hands' asset in the dataset side, instead of letting people run this pipeline.

Reply:
I'd say let's land this in the meantime. Could you message me at julietamartinez@meta.com with the images? I'll add them to the release package.

Reply:
I'd love to. Unfortunately, I do not have enough storage and resources to download the full datasets and run this tex_mean extraction for each ID locally. Therefore, it would be great if this could be done internally on your side.

Reply:
Got it. Have you run the code on these PRs on your side at all?

Reply:
Yes. the code is verified by training on AXE977 but fewer frames subset (100 frames).